Tasks/AzureRmWebAppDeployment/operations/PublishProfileUtility.ts

```diff
@@ -0,0 +1,53 @@
+import tl = require('vsts-task-lib/task');
+import { TaskParameters } from './TaskParameters';
+import fs = require('fs');
+import * as Constant from './Constants';
+
+var packageUtility = require('webdeployment-common/packageUtility.js');
+var parseString = require('xml2js').parseString;
+
+export class PublishProfileUtility {
+
+    private static async GetPublishProfileJsonFromFile(publishProfilePath: string): Promise<any> {
+        return new Promise((response, reject) => {
+            var pubxmlFile = packageUtility.PackageUtility.getPackagePath(publishProfilePath);
+            var publishProfileXML = fs.readFileSync(pubxmlFile);
+            parseString(publishProfileXML, (error, result) => {
+                if(!!error) {
+                    reject(error);
+                }
+                var propertyGroup = result && result.Project && result.Project.PropertyGroup ? result.Project.PropertyGroup : null;
+                if(propertyGroup) {
+                    for (var index in propertyGroup) {
+                        if (propertyGroup[index] && propertyGroup[index].WebPublishMethod[0] === "MSDeploy") {
+                            response(propertyGroup[index]);
+                        }
+                    }
+                }
+                reject(tl.loc('ErrorNoSuchDeployingMethodExists'));
+            });
+        });
+    }
+
+    public static async GetTaskParametersFromPublishProfileFile(taskParams: TaskParameters): Promise<any> {
+        try {
+            var publishProfileJs = await PublishProfileUtility.GetPublishProfileJsonFromFile(taskParams.PublishProfilePath);
+            if(!publishProfileJs.hasOwnProperty("MSDeployServiceURL")) {
+                throw new Error("MSDeployServiceURL not found in publish profile");
+            }
+            tl.debug("Publish Profile: " + publishProfileJs);
+            var msDeployPublishingProfile: any = {};
+            taskParams.WebAppName = (publishProfileJs.MSDeployServiceURL[0] as string).split(".")[0];
```

Comment:
Can you check pubxml file for slot apps?

Reply:
As per my understanding of publish profile, user have to maintain different publish profiles for slot, test and production environment and in each of the publish profile the MSDeployServiceURL will change as per environment.
https://social.msdn.microsoft.com/Forums/en-US/277909d3-72e5-43e6-ae2a-175bf085e6cf/how-to-deploy-to-a-different-slot-using-visual-studio-2015-publish-web-tool?forum=windowsazurewebsitespreview

https://stackoverflow.com/questions/33265898/using-web-deploy-how-do-you-publish-to-a-particular-slot